### PR TITLE
WTF-17482 - response caching support

### PR DESCRIPTION
**Pull request by carwallaby**
initially derped and merged to master. here is my re-attempt 🐧

### CODE REVIEW

File: test/unit/lib/index_test.js [hunk 1]
```diff
@@ -266,6 +266,45 @@ describe('Tagged API', function() {
...
+
+            it('optionally caches for custom amount of time', function() {
+                this.api.execute('floop', {}, {cache: 100});
+                this.clock.tick(100001);
+                this.api.execute('floop', {}, {cache: true});
+                this.clock.tick(1);
+                this.http.post.calledTwice.should.be.true;
+            });
+
+            it('does not duplicate calls if made before cache expiry time', function() {
```
Comment (djvirgen) [inline test/unit/lib/index_test.js]: > does not duplicate calls if made before cache expiry time

This description looks incorrect.

**Comment by djvirgen** [comment]
There is potential for a memory leak with the current implementation. We need to add some mechanism for automatically clearing expired cache entries. One of the simplest solutions is to delete all expired cache entries as a side effect of calling another method, like `TaggedAPI#execute`. However, it may not always be a cheap operation. One trick I've seen before is to use a random number generator to determine if the cache should be scrubbed, resulting in the cache being cleaned once every *n* calls. More advanced solutions involve using an LRU cache, but we don't need to go down that route right now.

File: test/unit/lib/index_test.js (hunk 1 above)
Comment (carwallaby) [inline test/unit/lib/index_test.js]: ok -- will reexamine~

**Comment by djvirgen** [comment]
🍒

**Pull request merged**